crates/wasmtime/src/runtime/vm.rs

```diff
@@ -336,6 +336,216 @@ impl ModuleRuntimeInfo {
     }
 }
 
+/// A number of bytes that's guaranteed to be aligned to the host page size.
+///
+/// This is used to manage page-aligned memory allocations.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct HostAlignedByteCount(
+    // Invariant: this is always a multiple of the host page size.
+    usize,
+);
+
+impl HostAlignedByteCount {
```

Comment:
Mind putting this type/impls/errors/etc in a dedicated submodule of the `vm` module?

Reply:
Done.